pennylane/measurements/vn_entanglement_entropy.py

```diff
@@ -0,0 +1,184 @@
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# pylint: disable=protected-access
+"""
+This module contains the qml.vn_entanglement_entropy measurement.
+"""
+from copy import copy
+from typing import Optional, Sequence
+
+import pennylane as qml
+from pennylane.wires import Wires
+
+from .measurements import StateMeasurement, VnEntanglementEntropy
+
+
+def vn_entanglement_entropy(wires0, wires1, log_base=None):
+    r"""Measures the `Von Neumann entanglement entropy <https://en.wikipedia.org/wiki/Entropy_of_entanglement>`_
+    of a quantum state:
+
+    .. math::
+
+        S(\rho_A) = -\text{Tr}[\rho_A \log \rho_A] = -\text{Tr}[\rho_B \log \rho_B] = S(\rho_B)
+
+    where :math:`S` is the Von Neumann entropy; :math:`\rho_A = \text{Tr}_B [\rho_{AB}]` and
+    :math:`\rho_B = \text{Tr}_A [\rho_{AB}]` are the reduced density matrices for each partition.
+    The Von Neumann entanglement entropy is a measure of the degree of quantum entanglement between
+    two subsystems constituting a pure bipartite quantum state. The entropy of entanglement is the
+    Von Neumann entropy of the reduced density matrix for any of the subsystems. If it is non-zero,
+    it indicates the two subsystems are entangled.
+
+    Args:
+        wires0 (Sequence[int] or int): the wires of the first subsystem
+        wires1 (Sequence[int] or int): the wires of the second subsystem
+        log_base (float): Base for the logarithm.
+
+    Returns:
+        VnEntanglementEntropyMP: measurement process instance
+
+    **Example:**
+
+    .. code-block:: python3
+
+        dev = qml.device("default.qubit", wires=2)
+        @qml.qnode(dev)
+        def circuit(x):
+            qml.RY(x, 0)
+            qml.Hadamard(0)
+            qml.CNOT([0, 1])
+            return qml.vn_entanglement_entropy([0], [1])
+
+    Executing this QNode:
+
+    >>> circuit(1.967)
+    0.16389850379003218
+
+    It is also possible to get the gradient of the previous QNode:
+
+    >>> param = np.array(np.pi/4, requires_grad=True)
+    >>> qml.grad(circuit)(param)
+    tensor(-0.62322524, requires_grad=True)
+
+    .. note::
+
+        Calculating the derivative of :func:`~pennylane.vn_entanglement_entropy` is currently supported when
+        using the classical backpropagation differentiation method (``diff_method="backprop"``)
+        with a compatible device and finite differences (``diff_method="finite-diff"``).
+
+    .. seealso:: :func:`~pennylane.vn_entropy` and :func:`pennylane.math.vn_entanglement_entropy`
+    """
+    wires0 = qml.wires.Wires(wires0)
+    wires1 = qml.wires.Wires(wires1)
+
+    # the subsystems cannot overlap
```

Comment:
So `qml.math.vn_entanglement_entropy` also checks that the two subsystems do not overlap. I'm not sure how it would affect jit compatibility.

Reply:
The check is only done if the wires are not abstract, so jitting should be fine.

Reply:
I mean the underlying math function that this measurement process calls in `process_density_matrix` also does this check, so I'm wondering if it would affect jit

Reply:
That's a good point. There are tests that test that both execution and differentiation works with jax jit, and they pass.